impeller/compiler/reflector.cc

```diff
@@ -667,32 +693,47 @@ std::vector<StructMember> Reflector::ReadStructMembers(
           member.columns == 1 &&           //
           member.vecsize == 1              //
       ) {
+        uint32_t stride = GetArrayStride<0>(struct_type, member, i);
+        if (stride == 0) {
+          stride = maybe_known_type.value().byte_size;
+        }
+        uint32_t element_padding = stride - maybe_known_type.value().byte_size;
         // Add the type directly.
         result.emplace_back(StructMember{
             maybe_known_type.value().name,         // type
             BaseTypeToString(member.basetype),     // basetype
             GetMemberNameAtIndex(struct_type, i),  // name
             struct_member_offset,                  // offset
-            maybe_known_type.value().byte_size     // byte_length
+            maybe_known_type.value().byte_size,    // size
+            stride * array_elements,               // byte_length
+            array_elements,                        // array_elements
+            element_padding,                       // element_padding
         });
-        current_byte_offset += maybe_known_type.value().byte_size;
+        current_byte_offset += stride * array_elements;
         continue;
       }
     }
 
     // Catch all for unknown types. Just add the necessary padding to the struct
     // and move on.
     {
-      const size_t byte_length =
-          (member.width * member.columns * member.vecsize) / 8u;
+      const size_t size = (member.width * member.columns * member.vecsize) / 8u;
+      uint32_t stride = GetArrayStride<0>(struct_type, member, i);
+      if (stride == 0) {
+        stride = size;
+      }
+      auto element_padding = stride - size;
       result.emplace_back(StructMember{
-          TypeNameWithPaddingOfSize(byte_length),  // type
-          BaseTypeToString(member.basetype),       // basetype
-          GetMemberNameAtIndex(struct_type, i),    // name
-          struct_member_offset,                    // offset
-          byte_length                              // byte_length
+          TypeNameWithPaddingOfSize(size),       // type
+          BaseTypeToString(member.basetype),     // basetype
+          GetMemberNameAtIndex(struct_type, i),  // name
+          struct_member_offset,                  // offset
+          size,                                  // size
+          stride * array_elements,               // byte_length
+          array_elements,                        // array_elements
+          0,                                     // element_padding
       });
-      current_byte_offset += byte_length;
+      current_byte_offset += stride * array_elements;
```

Comment:
nit: Did you mean to do `element_padding * array_elements`?

element_padding is unused

Reply:
Line 734 is commented as `element_padding`. Maybe that wasn't supposed to be `0`?

Reply:
Yup, 0 was only supposed to be used in the case below this one.